src/server/pps/server/api_server.go

```diff
@@ -1167,7 +1169,7 @@ func (a *apiServer) GarbageCollect(ctx context.Context, request *pps.GarbageColl
 	activeTags := make(map[string]bool)
 	for _, pipelineInfo := range pipelineInfos.PipelineInfo {
 		tags, err := objClient.ListTags(ctx, &pfs.ListTagsRequest{
-			Prefix:        client.HashPipelineID(pipelineInfo.ID),
```

Comment:
Same question here.  It concerns me whenever `ID` is replaced with `Name`, since the former changes when a pipeline is recreated while the latter doesn't.  Should we not instead use the `salt` here?